### ISSUE TITLE
Toon tanks memory leak

### ISSUE THREAD

**Issue by Triticum0** [cemu-project/Cemu#1265]
### Current Behavior

The game suffers from a  memory over time will crash the emulator

### Expected Behavior

Cemu not being 7GB  in size.

### Steps to Reproduce

1. Load game
2. start on the first mission
3. play until you start to see the memory balloon

### System Info (Optional)

OS: Windows 10
GPU: NVIDIA GeForce RTX 3060 Ti
CPU: AMD Ryzen 7 5700X 8-Core Processor

### Emulation Settings (Optional)

default settings

### Logs (Optional)

[Toon Tanks.txt](https://github.com/user-attachments/files/16322011/Toon.Tanks.txt)

**Comment by Squall-Leonhart** [comment]
Using memory is not the same thing as leaking memory, using 7GB is not unheard of in openworld titles, especially using OpenGL (Which you are)

**Comment by Exzap** [comment]
I can't replicate this. For me it stays around 1.4GB memory usage. Can you explain in more detail what you are doing in the game exactly (which missions?) and how fast memory usage increases

**Comment by Squall-Leonhart** [comment]
[an opengl memory leak was reported to occur between 472.12 and 536.40 when utilising glMaxShaderCompilerThreadsARB](https://forums.developer.nvidia.com/t/we-are-experiencing-like-a-memory-leak-using-the-glmaxshadercompilerthreadsarb/258442) in July 2023, it is possible this has been fixed post 545.

**Comment by Triticum0** [comment]
I updated to the latest drivers and retested on my Linux iso didn't prevent the memory leak.
Took 10 mins to get to 2.7 GB on Windows
on Linux got to close to 12 GB before opening Steam by mistake and Cemu dropped to 8 GB in a matter of seconds.
so it has nothing to do with the drivers on the original log or the os.

**Comment by Triticum0** [comment]
Also completed the game so will post a save file if needed, not sure if I can though. also affected more than just toon tanks so I don't think it's game-specific, more of a driver issue being my guess. If you need to give more information from the emulator just tell me what to do to help troubleshoot. Also didn't affect any emulated games in my testing

**Comment by Triticum0** [comment]
here the save if anyone needs it. 
[101a4400.tar.gz](https://github.com/user-attachments/files/16354895/101a4400.tar.gz)

**Comment by Exzap** [comment]
Did you try switching to Vulkan?

**Comment by Triticum0** [comment]
yes the Linux test as Vulkan is set by default on a fresh install

**Comment by Squall-Leonhart** [comment]
interestingly, the Eu version doesn't render with OpenGL, only with Vulkan

**Comment by Exzap** [comment]
Gonna chalk this up to a driver problem or something else outside of Cemu. I am unable to replicate this on any of my machines no matter if Windows or Linux.

**Comment by Triticum0** [comment]
I will retest then change compatibility report to playable

**Comment by Triticum0** [comment]
@Exzap Doing some more testing the memory leak only happens when entering new level after a while the memory stabilizes and will not increase any futher but once you change level or retry the mission the memory  will ramp up again. so only will happen when playing the game normally. Staying on one level will not cause the memory to increase indefinitely.  Cemu never does any garbage collection so on full play thought it will crash the game by running out of memory. I have another Graphic card to test it with. So if I can reproduce the issue on the rx9070 I would suggest you reopen the issue. Hopefully you can reproduce it and get fix as it happened on other game I tested as well.
